### PR TITLE
Move cleanup to a different container

### PR DESCRIPTION
**Pull request by raulcabello**
Move the cleanup controller and the code that's cleaning orphan `contents`,  `bundles` and `imageScans` to a different binary.

Run the fleet-cleanup in a different container

### CODE REVIEW

**Review by manno — APPROVED** [review]

File: internal/cmd/controller/cleanup/root.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"fmt"
+
+	command "github.com/rancher/fleet/internal/cmd"
+	"github.com/rancher/fleet/pkg/version"
+	"github.com/spf13/cobra"
+)
+
+var (
+	debugConfig command.DebugConfig
```
Comment (aruiz14) [inline internal/cmd/controller/cleanup/root.go:12]: This should not be needed, since the root command already configures it

File: go.sum [hunk 2]
```diff
@@ -820,6 +821,10 @@ github.com/rancher/helm/v3 v3.12.3-fleet1 h1:CCvZttHJChIG097BZIMYHX9rcuPFhgjmmnc
 github.com/rancher/helm/v3 v3.12.3-fleet1/go.mod h1:KPKQiX9IP5HX7o5YnnhViMnNuKiL/lJBVQ47GHe1R0k=
 github.com/rancher/lasso v0.0.0-20230830164424-d684fdeb6f29 h1:+kige/h8/LnzWgPjB5NUIHz/pWiW/lFpqcTUkN5uulY=
 github.com/rancher/lasso v0.0.0-20230830164424-d684fdeb6f29/go.mod h1:kgk9kJVMj9FIrrXU0iyM6u/9Je4bEjPImqswkTVaKsQ=
+github.com/rancher/wrangler v1.1.1 h1:wmqUwqc2M7ADfXnBCJTFkTB5ZREWpD78rnZMzmxwMvM=
+github.com/rancher/wrangler v1.1.1/go.mod h1:ioVbKupzcBOdzsl55MvEDN0R1wdGggj8iNCYGFI5JvM=
+github.com/rancher/wrangler-cli v0.0.0-20220624114648-479c5692ba22 h1:ADMwgJyVwmLXJBSm/nNobB1XGSmFCTA+TY/otxgIPu4=
+github.com/rancher/wrangler-cli v0.0.0-20220624114648-479c5692ba22/go.mod h1:vyO9SU60oplNFa5ZqoEAFWmYKgj2F6remdy8p6H0SgI=
```
Comment (aruiz14) [inline go.sum:827]: There must be something wrong. Please beware that we have removed `wrangler-cli` and `wrangler/v1` recently

**Review by aruiz14 — CHANGES_REQUESTED** [review]
Please review the comments before merging

File: internal/cmd/controller/cleanup/root.go (hunk 1 above)
Comment (manno) [reply internal/cmd/controller/cleanup/root.go:12]: That's because we merged something else first, this PR is in conflict.

Comment (raulcabello) [reply internal/cmd/controller/cleanup/root.go:12]: @manno is right. It has conflicts with https://github.com/rancher/fleet/pull/1899 which is the PR that changed the debug approach. I'm fixing the conflicts

File: go.sum (hunk 2 above)
Comment (raulcabello) [reply go.sum]: this was gone after fixing the conflicts with https://github.com/rancher/fleet/pull/1899

File: internal/cmd/controller/cleanup/root.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"fmt"
+
+	command "github.com/rancher/fleet/internal/cmd"
+	"github.com/rancher/fleet/pkg/version"
+	"github.com/spf13/cobra"
+)
+
+type CleanUp struct {
+	command.DebugConfig
```
Comment (aruiz14) [inline internal/cmd/controller/cleanup/root.go:12]: Doesn't `FleetManager` already includes a `DebugConfig`? Since it uses `PersistentPreRun`, I believe this is not needed.

Comment (raulcabello) [reply internal/cmd/controller/cleanup/root.go]: you're right! It's not needed to add it again in the subcomand. Removed in https://github.com/rancher/fleet/pull/1885/commits/8b1d535b72856e7501dd25512d867efcc9a5fe50

File: internal/cmd/controller/cleanup/root.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+func (c *CleanUp) Run(cmd *cobra.Command, args []string) error {
+	if c.Namespace == "" {
+		return fmt.Errorf("--namespace or env NAMESPACE is required to be set")
+	}
+	if err := start(cmd.Context(), c.Kubeconfig, c.Namespace); err != nil {
+		return err
+	}
+	<-cmd.Context().Done()
+
+	return nil
```
Comment (aruiz14) [inline internal/cmd/controller/cleanup/root.go:25]: Why is this needed? `RunOrDie` used in `start` is blocking, isn't it?
https://github.com/rancher/wrangler/blob/v2.1.2/pkg/leader/leader.go#L29

```suggestion
	return start(cmd.Context(), c.Kubeconfig, c.Namespace)
```

Comment (raulcabello) [reply internal/cmd/controller/cleanup/root.go:25]: I must have added it because I copied pasted from the [agent](https://github.com/rancher/fleet/blob/master/internal/cmd/agent/root.go#L48)

**Review by aruiz14 — APPROVED** [review]
